ibl_to_nwb/AlyxToNWB/io_tools.py

```diff
@@ -0,0 +1,208 @@
+        This method is current specific to units table to retrieve spike times for a given cluster
+        Parameters
+        ----------
+        datastring: str
+            comma separated dataset names ex: "spike.times,spikes.clusters"
+        Returns
+        -------
+        ls_merged: [list, None]
+            list of length number of probes. Each element is a list > each element is an array of cluster's spike times
+        """
+        spike_clusters, spike_times = datastring.split(',')
+        if spike_clusters not in self.loaded_datasets.keys():
+            spike_cluster_data = self.one_object.load(self.eid, dataset_types=[spike_clusters])
+            self.loaded_datasets.update({spike_clusters: spike_cluster_data})
+        else:
+            spike_cluster_data = self.loaded_datasets[spike_clusters]
+        if spike_times not in self.loaded_datasets.keys():
+            spike_times_data = self.one_object.load(self.eid, dataset_types=[spike_times])
+            self.loaded_datasets.update({spike_times: spike_times_data})
+        else:
+            spike_times_data = self.loaded_datasets[spike_times]
+        if not ((spike_cluster_data is None) | (spike_cluster_data is None)):  # if bot hdata are found only then
+            ls_merged = []
+            if not self.data_attrs_dump.get(
+                    'unit_table_length'):  # if unit table length is not known, ignore spike times
+                return None
+            if np.abs(np.max(spike_cluster_data[0]) - self.data_attrs_dump['unit_table_length'][0]) > 20:
+                i_loop = np.arange(self.no_probes - 1, -1, -1)
+            else:
+                i_loop = np.arange(self.no_probes)
+            for j, i in enumerate(i_loop):
+                df = pd.DataFrame({'sp_cluster': spike_cluster_data[i], 'sp_times': spike_times_data[i]})
+                data = df.groupby(['sp_cluster'])['sp_times'].apply(np.array).reset_index(name='sp_times_group')
+                ls_grouped = [[np.nan]]*self.data_attrs_dump['unit_table_length'][
+                    j]  # default spiking time for clusters with no time
+                for index, sp_list in data.values:
+                    ls_grouped[index] = sp_list
+                ls_merged.extend(ls_grouped)
+            return ls_merged
+
+    def _load_as_array(self, dataset_to_load, dataset_key, loaded_dataset_):
+        """
+        Takes variable data formats: .csv, .npy, .bin, .meta, .json and converts them to ndarray.
+        Parameters
+        ----------
+        loaded_dataset_: [SessionDataInfo]
+        Returns
+        -------
+        out_data: [ndarray, list]
+        """
+        if len(loaded_dataset_.data) == 0 or loaded_dataset_.data[0] is None:  # dataset not found in the database
+            return None
+        datatype = [i.suffix for i in loaded_dataset_.local_path]
+        dataloc = [i for i in loaded_dataset_.local_path]
+
+        if datatype[-1] in ['.csv', '.npy']:  # csv is for clusters metrics
```

Comment:
You should call different loading functions depending on if the data is .csv or .npy